### DISCUSSION TITLE
[v6.0.4] Issue with generating migration

### DISCUSSION THREAD

**Discussion by tsvetann** [mikro-orm/mikro-orm#5149] [Q&A]
hello,

I have an Entity from which I would like to create a migration file. Here is my Entity:

```
@Entity()
export class UserLikes {

  @Property()
  first_name!: string;

  constructor(first_name: string) {
    super();
    this.first_name = first_name;
  }
}
```

I programatically run `orm.migrator.createMigration(path, false, false, 'test')` and I get the following migration created:

```
async up(): Promise<void> {
    this.addSql('create table "user_likes" ......');
  }

  async down(): Promise<void> {
    this.addSql('create table "comment" ("comment_id" serial primary key, .....');
    this.addSql('create index "comment_entity_index" on "comment" ("entity");');
    this.addSql('create index "comment_parent_comment_id_index" on "comment" ("parent_comment_id");');
    this.addSql('CREATE INDEX comment_user_index ON public.comment USING btree ("user");');
  }
```

Why is it adding this extra diff in the down() method? I do have another table called `comment` already generated and applied via migrations but I must have misconfigured something to get the comment's table stuff here. I would only expect to have the user_likes table dropped in the down() method. Thx

**Comment by B4nan** [comment]
Maybe an outdated snapshot file? Try removing it.

**Reply by tsvetann** [reply]
@B4nan I should've mentioned that I have disabled snapshots. Is that the reason?

**Reply by B4nan** [reply]
Is your schema in sync or do you have pending migrations?

**Reply by tsvetann** [reply]
No pending migrations. I have two Entities -> generated two migrations -> applied the migrations -> present in DB. Then created a third Entity -> generated migration file from the entity -> got the unexpected result above. Any ideas how I can debug this further? I would prefer to keep snapshots off

**Reply by B4nan** [reply]
> generated two migrations -> applied the migrations

You cant generate two migrations without snapshots like that, you generate one, execute, then generate the other. Is that what you mean? Because for what you said you need the snapshot.

> I would prefer to keep snapshots off

Why?

**Reply by tsvetann** [reply]
Correct, I generated the migration files one by one. 

I would like to keep the snapshots off due to the complexity of the application. Entities and migration files are to be generated in different folders and also from npm packages so I am not sure how to handle all of it with a single snapshot file. The documentation on snapshots is only a paragraph and touches only the very basic use case.

**Reply by tsvetann** [reply]
@B4nan just tried with snapshot enabled. Same issue

**Reply by tsvetann** [reply]
After a bit more testing I discovered that every new migration created, replaces the data in the snapshot (meaning the prev table snapshot is gone)

**Reply by B4nan** [reply]
That's exactly the point of snapshots, they represent the latest state, after the latest migration (even if it wasn't executed yet).

**Reply by tsvetann** [reply]
My question is: I create a comment entity -> generate migration -> snapshot created reflecting comment schema. Next I create user entity -> generate migration -> comment schema is removed from snapshot and use schema becomes the new snapshot. Why is the comment schema deleted from the snapshot (which adds removing the comment table in the user migration file). I would like to keep both. Is this a misconfiguration?

**Reply by B4nan** [reply]
Nothing should be "removed from snapshot", the snapshot is literally your whole metadata collection. If you are missing some entity in it, it means it wasn't discovered.

Hard to help without seeing any details, so far it was all just words, so I don't really know what's happening on your end. But what you say now seems rather impossible.

**Comment by tsvetann** [comment]
> Nothing should be "removed from snapshot", the snapshot is literally your whole metadata collection. If you are missing some entity in it, it means it wasn't discovered.
> 
> Hard to help without seeing any details, so far it was all just words, so I don't really know what's happening on your end. But what you say now seems rather impossible.

@B4nan thanks for your help and patience. I think I discovered what the issue is. Since I have migration files in different directories, there is no way for me to have a single snapshotPath as it is not configurable, hence I had the snapshot disabled. Are you planning to add the snapshotPath as part of the migrations config?

**Reply by B4nan** [reply]
I think you should be able to put an absolute path to the `migrations.snapshotName`.
